File: android/src/main/kotlin/dev/hebei/pag/PAGViewImpl.kt
package dev.hebei.pag

import android.content.Context
import org.libpag.PAGComposition
import org.libpag.PAGView

class PAGViewImpl(registrar: PAGApiPigeonProxyApiRegistrar, private val context: Context) :
    PigeonApiPAGViewApi(registrar) {
    override fun pigeon_defaultConstructor(): PAGView {
        return PAGView(context)
    }

    override fun getComposition(pigeon_instance: PAGView): PAGComposition {
        return pigeon_instance.composition
    }

    override fun setCompositon(pigeon_instance: PAGView, newComposition: PAGComposition) {
        pigeon_instance.composition = newComposition
    }

    override fun repeatCount(pigeon_instance: PAGView): Long {
        return pigeon_instance.repeatCount().api
    }

    override fun setRepeatCount(pigeon_instance: PAGView, repeatCount: Long) {
        pigeon_instance.setRepeatCount(repeatCount.impl)
    }

    override fun scaleMode(pigeon_instance: PAGView): PAGScaleModeApi {
        return pigeon_instance.scaleMode().scaleModeApi
    }

    override fun setScaleMode(pigeon_instance: PAGView, value: PAGScaleModeApi) {
        pigeon_instance.setScaleMode(value.impl)
    }

    override fun getProgress(pigeon_instance: PAGView): Double {
        return pigeon_instance.progress
    }

    override fun setProgress(pigeon_instance: PAGView, value: Double) {
        pigeon_instance.progress = value
    }

    override fun sync(pigeon_instance: PAGView): Boolean {
        return pigeon_instance.isSync
    }

    override fun setSync(pigeon_instance: PAGView, value: Boolean) {
        pigeon_instance.isSync = value
    }

    override fun isPlaying(pigeon_instance: PAGView): Boolean {
        return pigeon_instance.isPlaying
    }

    override fun play(pigeon_instance: PAGView) {
        pigeon_instance.play()
    }

    override fun pause(pigeon_instance: PAGView) {
        pigeon_instance.pause()
    }

    override fun stop(pigeon_instance: PAGView) {
        pigeon_instance.stop()
    }
}

File: pag_darwin/lib/src/pag_view_impl.dart
import 'package:flutter/services.dart';
import 'package:flutter/widgets.dart';
import 'package:pag_platform_interface/pag_platform_interface.dart';

import 'api.g.dart';
import 'api.x.dart';

const _kViewType = 'hebei.dev/PAGView';

abstract final class PAGViewImpl extends PAGView {
  PAGViewApi api;

  factory PAGViewImpl.iOS() => _PAGiOSViewImpl();
  factory PAGViewImpl.macOS() => _PAGmacOSViewImpl();

  PAGViewImpl()
      : api = PAGViewApi(),
        super.impl();

  @override
  Future<PAGComposition> getComposition() async {
    final value = await api.getComposition();
    return value.impl;
  }

  @override
  Future<void> setCompositon(PAGComposition value) async {
    await api.setCompositon(value.api);
  }

  @override
  Future<int> getRepeatCount() async {
    final value = api.getRepeatCount();
    return value;
  }

  @override
  Future<void> setRepeatCount(int value) async {
    await api.setRepeatCount(value);
  }

  @override
  Future<PAGScaleMode> getScaleMode() async {
    final value = await api.getScaleMode();
    return value.impl;
  }

  @override
  Future<void> setScaleMode(PAGScaleMode value) async {
    await api.setScaleMode(value.api);
  }

  @override
  Future<double> getProgress() async {
    final value = await api.getProgress();
    return value;
  }

  @override
  Future<void> setProgress(double value) async {
    await api.setProgress(value);
  }

  @override
  Future<bool> isPlaying() async {
    final value = await api.isPlaying();
    return value;
  }

  @override
  Future<void> play() async {
    await api.play();
  }

  @override
  Future<void> pause() async {
    await api.pause();
  }

  @override
  Future<void> stop() async {
    await api.stop();
  }
}

final class _PAGiOSViewImpl extends PAGViewImpl {
  @override
  Widget build(BuildContext context) {
    final identifier = api.pigeon_instanceManager.getIdentifier(api);
    return UiKitView(
      viewType: _kViewType,
      layoutDirection: TextDirection.ltr,
      creationParams: identifier,
      creationParamsCodec: const StandardMessageCodec(),
    );
  }
}

final class _PAGmacOSViewImpl extends PAGViewImpl {
  @override
  Widget build(BuildContext context) {
    final identifier = api.pigeon_instanceManager.getIdentifier(api);
    return AppKitView(
      viewType: _kViewType,
      layoutDirection: TextDirection.ltr,
      creationParams: identifier,
      creationParamsCodec: const StandardMessageCodec(),
    );
  }
}
